client/app/shared/order.service.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Observable, throwError as observableThrowError } from "rxjs";
+import { catchError, map } from "rxjs/operators";
+import { Order } from "./order.model";
+
+@Injectable({
+    providedIn: "root"
+})
+export class OrderService {
+
+    constructor(private http: HttpClient) {
+    }
+
+    getOrders(): Observable<any> {
+        return this.http.get("/api/orders/in-progress")
+            .pipe(
+                map((response: any) => this.toOrderArray(response)),
+                catchError((error: any) => observableThrowError(error))
+            );
+    }
+
+
+    toOrderArray(response: any): Order[] {
```

Comment:
I'm not sure this function is necessary here. getOrders() already gives you the response and you don't usually have functions like this in the service. 

The service usually only has functions that call the API. You can use the model in getOrders(), like this: 
`map((response: Order) => OrderFactory.create(order)` if you want to map it and have it strictly typed but we don't really need to do it this way I think... 

The tutor will probably not even look into these details... 

@last1here @jessicawong217 @Nadiajayy What do you guys think?

Reply:
To be honest I've just copied how we do it in work! Happy to change it if it's not necessary :)

Reply:
Could also cast it within the function  if we dont plan to add methods to the Order `return this.http.get("/api/orders/in-progress") as Observable<Order[]>` or within the pipe. Or just set the return type of the function `getOrders(): Observable<Order[]>` saves the extra loop and assignments. 

Would need to change the types `id` to `_id` to match the response.